harvester_e2e_tests/apis/test_support_bundle.py

```diff
@@ -0,0 +1,109 @@
+# Copyright (c) 2022 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of version 3 of the GNU General Public License as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE about this file by physical or electronic mail,
+# you may find current contact information at www.suse.com
+
+import json
+import os
+import tempfile
+import zipfile
+
+import polling2
+import pytest
+from harvester_e2e_tests import utils
+
+pytest_plugins = [
+    'harvester_e2e_tests.fixtures.api_endpoints',
+    'harvester_e2e_tests.fixtures.user',
+    'harvester_e2e_tests.fixtures.session',
+    'harvester_e2e_tests.fixtures.support_bundle'
+]
+
+
+def test_create_volume_missing_description(
+        admin_session, harvester_api_endpoints):
+    request_json = utils.get_json_object_from_template('support_bundle')
+    del request_json['spec']['description']
+    resp = admin_session.post(
+        harvester_api_endpoints.create_support_bundle, json=request_json)
+    assert resp.status_code == 422
+    assert 'is invalid: spec.description: Required value' in resp.text
+
+
+@pytest.mark.p1
+@pytest.mark.slow
+def test_creates_successful_bundle_and_downloads(
+        admin_session, harvester_api_endpoints):
+    """
+    SupportBundle generic
+    Covers:
+        The support bundle creating
+        The support bundle generation and progress, progressing
+        The support bundle being able to be downloaded
+        The file structure existing in the support bundle correctly in logs
+    """
+    request_json = utils.get_json_object_from_template('support_bundle')
+    resp = admin_session.post(
+        harvester_api_endpoints.create_support_bundle, json=request_json)
+    assert resp.status_code == 201
+    resp_json = json.loads(resp.text)
+    name_to_use_for_download = resp_json.get('metadata').get('name')
+    progress = 0
+
+    def _wait_for_progress_to_be_ready():
+        nonlocal progress
+        resp_progress_check = admin_session.get(
+            harvester_api_endpoints.view_support_bundle +
+            name_to_use_for_download)
+        json_progress_result = json.loads(resp_progress_check.text)
+        progress = json_progress_result.get('status', {}).get('progress', 0)
+        if progress != 100:
+            return False
+        return True
+
+    successful_support_bundle_generation = polling2.poll(
+        _wait_for_progress_to_be_ready,
+        step=5,
+        timeout=300
+    )
+    assert successful_support_bundle_generation, ('Timed out waiting for ' +
```

Comment:
the addition of the `+` was to make the pep8 linter happy, to create a wrap because of [E501, for flake8/pep8/autopep linting](https://www.flake8rules.com/rules/E501.html) - I can refactor to remove that :smile: